Create secret for bind credentials

djzager opened the pull request:
When `asb_encode_binding` is used, we want for a secret to be created in
the transient namespace where the APB is being executed.

See [the proposal](https://github.com/openshift/ansible-service-broker/blob/master/docs/proposals/prop-apb-gen-creds.md)
for more information.

djzager committed:
Create secret for bind credentials

When `asb_encode_binding` is used, we want for a secret to be created in
the transient namespace where the APB is being executed.

See [the
proposal](https://github.com/openshift/ansible-service-broker/blob/master/docs/proposals/prop-apb-gen-creds.md)
for more information.

fabianvf commented on library/asb_encode_binding.py:
```diff
@@ -44,9 +44,18 @@
+    config.load_kube_config()
+    api = client.CoreV1Api()
+except Exception as error:
+    ansible_module.fail_json(msg="Error attempting to load kubernetes client: " + str(error))
```
better to use python string formatting here, ex:  `"Error is {}".format(error)`

fabianvf commented on library/asb_encode_binding.py:
```diff
@@ -44,9 +44,18 @@
+    ansible_module.fail_json(msg="Error attempting to load kubernetes client: " + str(error))
 
 ENCODED_BINDING_PATH = "/var/tmp/bind-creds"
+ENV_NAMESPACE = "POD_NAMESPACE"
```
why is this set to `POD_NAMESPACE`?

djzager replied:
https://github.com/openshift/ansible-service-broker/blob/master/pkg/apb/executor.go#L137

djzager committed:
Use python string formatting

cfchase reviewed (approved):
ACK

djzager committed:
Encode binding as secret named after pod's name

Giving the secret a name equivalent to the pod's name is cleaner with
respect to the broker knowing how to find the secret.
